cwms-data-api/src/test/java/cwms/cda/data/dao/StreamLocationDaoTestIT.java

```diff
@@ -0,0 +1,260 @@
+    private static final List<Location> LOCATIONS_CREATED = new ArrayList<>();
+    private static final List<String> STREAM_LOC_IDS = new ArrayList<>();
+    private static final List<Stream> STREAMS_CREATED = new ArrayList<>();
+
+    @BeforeAll
+    public static void setup() {
+        try {
+            for (int i = 0; i < 2; i++) {
+                String testLoc = "STREAM_LOC_" + System.currentTimeMillis()/10;
+                STREAM_LOC_IDS.add(testLoc);
+                createAndStoreTestLocation(testLoc, "STREAM_LOCATION");
+                createAndStoreTestStream(testLoc + "S");
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void createAndStoreTestStream(String testLoc) throws SQLException {
+        createAndStoreTestLocation(testLoc, "STREAM");
+        CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+        db.connection(c-> {
+            StreamDao streamDao = new StreamDao(getDslContext(c, OFFICE_ID));
+            Stream streamToStore = new Stream.Builder()
+                    .withId(new CwmsId.Builder()
+                            .withOfficeId(OFFICE_ID)
+                            .withName(testLoc)
+                            .build())
+                    .withLength(100.0)
+                    .withLengthUnits("km")
+                    .build();
+            STREAMS_CREATED.add(streamToStore);
+            streamDao.storeStream(streamToStore, true);
+        });
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        for(Location location : LOCATIONS_CREATED){
+            try {
+                CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+                db.connection(c-> {
+                    LocationsDao locationsDao = new LocationsDaoImpl(getDslContext(c, OFFICE_ID));
+                    locationsDao.deleteLocation(location.getName(), location.getOfficeId(), true);
+                });
+            } catch(SQLException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+        LOCATIONS_CREATED.clear();
+
+        for(Stream stream : STREAMS_CREATED){
+            try {
+                CwmsDatabaseContainer<?> db = CwmsDataApiSetupCallback.getDatabaseLink();
+                db.connection(c-> {
+                    StreamDao streamDao = new StreamDao(getDslContext(c, OFFICE_ID));
+                    streamDao.deleteStream(stream.getId().getOfficeId(), stream.getId().getName(), DeleteRule.DELETE_ALL);
+                });
+            } catch(SQLException ex) {
+                throw new RuntimeException(ex);
+            }
+        }
+        STREAMS_CREATED.clear();
+        STREAM_LOC_IDS.clear();
+    }
+
+    @Test
+    void testRoundTrip() throws Exception {
+        CwmsDatabaseContainer<?> databaseLink = CwmsDataApiSetupCallback.getDatabaseLink();
+        databaseLink.connection(c -> {
```

Comment:
See above, should be done as the "webuser".

Reply:
passed in webUser param